### DISCUSSION TITLE
Struggling with JWE Decryption

### DISCUSSION THREAD

**Discussion by itsupply25** [panva/jose#434] [Q&A]
Hello, 

I'm trying to decrypt an JWE Token. But I'm struggling. 
I've got the JWE Token and a secret to decode. How is the correct usage of jose, to decode this?

header: "alg": "A256KW", "enc": "A256CBC-HS512", "typ": "JWT"

Example Token:
eyJhbGciOiJBMjU2S1ciLCJlbmMiOiJBMjU2Q0JDLUhTNTEyIiwidHlwIjoiSldUIn0.8lk3HXEJ5uvApko4umGH1_rLhdl-x7d-81HeFCz8I-1idij5crpQ1tQR7vVCxFVqhRN0B99NHlbHfhiSWKY2gmzMqWgDt56u.WuietiLj1lZvezkBE1ZvrA.VXc4EJ2sjQYh5D3Jf_bANTcwPe8lddJs9KsID0ERHUeU18nP0Hc0pEae_L4s7R-bGQbLmfUfkQwxO7AI9e8s2WwEHLrk0oYWU_LEV-xvTu_M-WH2gfMExDHcf08tEw0gfZsAf_BxRmwh74Ng-voscfT5tA7u7MJb0lKvZwvB5f_94CuWMvmVoKlerASBA2uH6J3554ST1suhgnKmGdIPPOuy2lQaXK6c0h8BHsTEG38tFoMxfGcP8ZQGxUCpzEvxt-amVQOgydNFgg9uxOg1MYRTdbNB8brsLG106dL_RNBkzwEKKXs-_Z36PjyR_DrP044w1H_SOqyJcReC-rkTw5Kh9IQIDalqrPpvQblce9hjm_qDJjhobMH2DxZqejJ4cizkbkD6kRvRR716yy17lsTeWkUoIJmQgp9Y2b2wogU.qbFmjV7paINPxs1l5hUHo9uTUGMfAIwVtWcCKRqyva0

Decrypt symmetric key:
Vaguely-Fresh-Orbit-0354-0314-81

I wasted a lot of hours, on this. And also with lots of browsing the internet I do not make progress with this topic. 

Thanks in advance for your help!

Best regards

**Comment by panva** [comment]
https://github.com/panva/jose/blob/v4.8.3/docs/functions/jwt_decrypt.jwtDecrypt.md#readme

Use the example in the JWT Decrypt function, substitute your encrypted JWT and for `secretKey` use `new TextEncoder().encode('Vaguely-Fresh-Orbit-0354-0314-81')`.

The token you pasted is expired now, so the decryption will succeed but subsequent claim validation will rightly fail.

This is the parsed encrypted claims set.

```js
{
  nameid: '0',
  unique_name: '123546789',
  claimRefTenantDomain: 'testserver.com',
  claimIsAdmin: '1',
  given_name: 'Max',
  family_name: 'Mustermann',
  gender: 'm',
  roles: '1,26,51',
  roles_codes: 'Hugo,Karl,Mike',
  nbf: 1659630043,
  exp: 1659630103,
  iat: 1659630043,
  iss: 'issuer.testserver.net',
  aud: 'testserver.com '
}
```

**Reply by itsupply25** [reply]
Thank you for your answert. It seems to work! How can I handle it, if parsed claim is decoded additionally. Then I get an error. Do I have to use Option inflateRaw? Do you have an example for additionally JWT decode?

Thanks in advance. 

Best regards
